bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/servlet/HomeConnectServlet.java

```diff
@@ -0,0 +1,629 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.WebContext;
+import org.thymeleaf.extras.java8time.dialect.Java8TimeDialect;
+import org.thymeleaf.templatemode.TemplateMode;
+import org.thymeleaf.templateresolver.ServletContextTemplateResolver;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializer;
+
+/**
+ *
+ * Home Connect servlet.
+ *
+ * @author Jonas Brüstel - Initial Contribution
+ */
+@NonNullByDefault
+@Component(service = HomeConnectServlet.class, scope = ServiceScope.SINGLETON, immediate = true)
+public class HomeConnectServlet extends HttpServlet {
+
+    private static final String SLASH = "/";
+    private static final String SERVLET_NAME = "homeconnect";
+    private static final String SERVLET_PATH = SLASH + SERVLET_NAME;
+    private static final String ASSETS_PATH = SERVLET_PATH + "/asset";
+    private static final String ROOT_PATH = SLASH;
+    private static final String APPLIANCES_PATH = "/appliances";
+    private static final String REQUEST_LOG_PATH = "/log/requests";
+    private static final String EVENT_LOG_PATH = "/log/events";
+    private static final String REQUEST_COUNT_PATH = "/requests";
+    private static final String DEFAULT_CONTENT_TYPE = "text/html; charset=UTF-8";
+    private static final String PARAM_CODE = "code";
+    private static final String PARAM_STATE = "state";
+    private static final String PARAM_EXPORT = "export";
+    private static final String PARAM_ACTION = "action";
+    private static final String PARAM_BRIDGE_ID = "bridgeId";
+    private static final String PARAM_THING_ID = "thingId";
+    private static final String PARAM_PATH = "path";
+    private static final String ACTION_AUTHORIZE = "authorize";
+    private static final String ACTION_CLEAR_CREDENTIALS = "clearCredentials";
+    private static final String ACTION_SHOW_DETAILS = "show-details";
+    private static final String ACTION_ALL_PROGRAMS = "all-programs";
+    private static final String ACTION_AVAILABLE_PROGRAMS = "available-programs";
+    private static final String ACTION_SELECTED_PROGRAM = "selected-program";
+    private static final String ACTION_ACTIVE_PROGRAM = "active-program";
+    private static final String ACTION_OPERATION_STATE = "operation-state";
+    private static final String ACTION_POWER_STATE = "power-state";
+    private static final String ACTION_DOOR_STATE = "door-state";
+    private static final String ACTION_REMOTE_START_ALLOWED = "remote-control-start-allowed";
+    private static final String ACTION_REMOTE_CONTROL_ACTIVE = "remote-control-active";
+    private static final String ACTION_PUT_RAW = "put-raw";
+    private static final String ACTION_GET_RAW = "get-raw";
+    private static final DateTimeFormatter FILE_EXPORT_DTF = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH-mm");
```

Comment:
Why not ISO8601?

Reply:
Will change it to `ISO_OFFSET_DATE_TIME`